src/components/TodoList.jsx

```diff
@@ -13,51 +13,121 @@ class TodoListXP extends React.Component {
+            }
+        }
+        if (option === "Newer") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.date - task2.date))).reverse()
+            }
+        }
+        if (option === "Older") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.date - task2.date))
+            }
+        }
 
-        const newState = {
+        this.setState(newState)
+    }
+
+
+    taskTexts = () => {
+        const taskNames = []
+        for (const task of this.state.tasks) {
+            taskNames.push(task.taskName)
+        }
+        return taskNames
+    }
+
+
+    changeText = (text, newText) => {
+        const { tasks } = this.state
+        const id = this.taskTexts().indexOf(text)
+        tasks[id].taskName = newText
+        this.setState({
             ...this.state,
-            tasks: tasks.filter(elem => elem != text)
+            tasks: tasks
+        })
+    }
+
+
+    removeTask(text) {
+        const { tasks } = this.state
+        tasks.splice(this.taskTexts().indexOf(text), 1)
+        const newState = {
+            ...this.state
         }
         this.setState(newState);
     }
 
     addTask() {
         const { tasks } = this.state;
-
         this.setState({
             input: "",
-            tasks: tasks.concat(this.state.input)
+            tasks: tasks.concat({ taskName: this.state.input, date: new Date() })
         });
     }
 
+
+
     onInputChange(event) {
         const text = event.target.value;
-        this.setState({ ...this.state, input: text });
+        this.setState({ ...this.state, input: text })
+    }
+
+    removeAllTasks() {
+        this.setState(
+            {
+                tasks: []
+            }
+        )
     }
 
     render() {
         return <div>
             <div>
+                <select onChange={this.sortTasks}>
+                    <option>Sort by</option>
+                    <option>Newer</option>
+                    <option>Older</option>
+                    <option>Name asc</option>
+                    <option>Name desc</option>
+                </select>
                 <button onClick={this.addTask}>Add Task</button>
                 <input type="text" value={this.state.input} onChange={this.onInputChange}></input>
+                <button onClick={this.removeAllTasks}>Remove Tasks</button>
             </div>
-            <ul style={todoStyle}>{
-                this.state.tasks.map(text => <TodoPoint key={text} todoText={text} removeTask={this.removeTask} />)
+            <ul id="todoPoints" style={todoStyle} >{
+                this.state.tasks.map(text => <TodoPoint key={this.state.tasks.indexOf(text)} todoText={text.taskName} removeTask={this.removeTask} todoDate={text.date.toUTCString().slice(0, 22)} changeText={this.changeText} oneInEdit={this.state.bIsOneTaskInEdit} />)
```

Comment:
This is not readable. Lemme show u an example of how to solve this problem:

![image](https://user-images.githubusercontent.com/3951613/88214641-2bfc3900-cc63-11ea-86ea-50ddaa79b5b2.png)

![image](https://user-images.githubusercontent.com/3951613/88214672-374f6480-cc63-11ea-8a72-2a1774a34542.png)


Notice how I extracted a long cardStyle property away from JSX and put it above the `return` statement of `render()`. Do this as well for long properties or attributes. You want your JSX as clean as possible, cause JSX is messy by default. We want ourselves and others to be able to read it :)